corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py

```diff
@@ -1671,10 +1762,10 @@ def testOtherUserReassignsIndexed(self):
         # hacky
         self.assertTrue("something new" in payload)
         self.assertTrue("hi!" in payload)
-        # also check that the latest sync log knows those cases are no longer relevant to the phone
-        log = synclog_from_restore_payload(payload)
-        self.assertFalse(log.phone_is_holding_case(case_id))
-        self.assertFalse(log.phone_is_holding_case(parent_id))
+        # also check that they are not sent to the phone on next sync
+        next_sync_log = self.get_next_sync_log(restore_id=latest_sync_log.get_id, version=V2)
+        assert_user_doesnt_have_cases(self, self.user, [case_id, parent_id],
+            restore_id=next_sync_log.get_id)
```

Comment:
This was changed to use `assert_user_doesnt_have_case`, which performs a sync before doing the check. This was especially confusing to me because `check_payload_has_case_ids` verifies that the case ids are in the payload, and then just below that it was verifying that `log.phone_is_holding_case` returned false for the same ids. That doesn't even sound possible for a single sync log/payload. Maybe I'm not understanding something about how this worked before?